src/staking.cairo

```diff
@@ -0,0 +1,77 @@
+#[starknet::contract]
+mod Staking {
+    use starknet::{ContractAddress, Zeroable};
+    use governance::traits::{
+        IStaking, IStakingDispatcher, IStakingDispatcherTrait, IERC20Dispatcher,
+        IERC20DispatcherTrait
+    };
+
+    #[derive(starknet::Event, Drop)]
+    #[event]
+    enum Event {
+        Staked: Staked,
+        Unstaked: Unstaked,
+        Increased_Lock_Time: Increased_Lock_Time,
+        CARM_Address_Set: CARM_Address_Set,
+        veCARM_Address_Set: veCARM_Address_Set
+    }
+
+    #[derive(starknet::Event, Drop)]
+    struct Staked {
+        user: ContractAddress,
+        CARM_amount_staked: u128,
+        veCARM_received: u128,
+        lock_period: felt252
+    }
+
+    #[derive(starknet::Event, Drop)]
+    struct Unstaked {
+        user: ContractAddress,
+        CARM_amount_unstaked: u128,
+        total_veCARM_received: u128,
+    }
+
+    #[derive(starknet::Event, Drop)]
+    struct Increased_Lock_Time {
+        user: ContractAddress,
+        new_lock_period: felt252
+    }
+
+    #[derive(starknet::Event, Drop)]
+    struct CARM_Address_Set {
+        owner: ContractAddress,
+        CARM_token_address: ContractAddress
+    }
+
+    #[derive(starknet::Event, Drop)]
+    struct veCARM_Address_Set {
+        owner: ContractAddress,
+        veCARM_token_address: ContractAddress
+    }
```

Comment:
I don't think these are necessary. Also, no need to emit the owner in this case.

Reply:
Cool, okay. I will make necessary adjustments and start off